app/src/components/assets/Asset.tsx

```diff
@@ -0,0 +1,165 @@
+import { faBan, faFlag } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IonCard, IonCardContent, IonCardHeader, IonCardSubtitle, IonCardTitle, IonCol, IonContent, IonItem, IonLabel, IonList, IonRow, useIonViewWillLeave } from "@ionic/react";
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import Api from "../../controllers/Api";
+import { s3url, fileExtensionToIcon, formatSize } from "../../globals/functions";
+import { AssetData, AssetTypeData } from "../../globals/interfaces";
+import Page from "../../pages/Page";
+
+const Asset = () => {
+    let { type, asset } = useParams<{type: string, asset: string}>();
```

Comment:
Could we not do this by creating a new instance of the interface?

Reply:
Casting with < > feels a bit hacky

Reply:
This was based on guides for params found in react guidance, happy to change if there is a better way that you know!